File: src/main/java/com/gxsneaker/gxsneaker/service/CoGiayService.java
package com.gxsneaker.gxsneaker.service;

import com.gxsneaker.gxsneaker.dto.CoGiayDTO;

import java.util.List;

public interface CoGiayService {
    List<CoGiayDTO> getAll();

    CoGiayDTO getById(Long id);

    CoGiayDTO create(CoGiayDTO dto);

    CoGiayDTO update(Long id, CoGiayDTO dto);

    void delete(Long id);
}

File: gx-sneaker-ui/src/services/coGiayService.js
import axios from "axios"

const API = "http://localhost:8080/api/co-giay"

export const getAll = () => axios.get(API)
export const getById = (id) => axios.get(`${API}/${id}`)
export const create = (data) => axios.post(API, data)
export const update = (id, data) => axios.put(`${API}/${id}`, data)
export const remove = (id) => axios.delete(`${API}/${id}`)
